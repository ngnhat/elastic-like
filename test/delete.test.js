/**
 * Created by ngnhat on Sat May 25 2019
 */
const Store = require('../index');

describe('single field:', () => {
  it('single doc - single term', () => {
    const store = new Store();

    store.add({ Id: 1, Code: 'aaa', Name: 'quick brown fox' });
    store.delete(1);

    expect(store.search('fox')).toEqual([]);
  });

  it('multiple doc - single term', () => {
    const store1 = new Store();
    const store2 = new Store();

    store1.add({ Id: 1, Code: '1', Name: 'The quick brown fox' });
    store1.add({ Id: 2, Code: '2', Name: 'The quick brown fox jumps over the lazy dog' });
    store1.add({ Id: 4, Code: '4', Name: 'Brown fox hahaha brown dog' });

    store2.add({ Id: 1, Code: '1', Name: 'The quick brown fox' });
    store2.add({ Id: 2, Code: '2', Name: 'The quick brown fox jumps over the lazy dog' });
    store2.add({ Id: 3, Code: '3', Name: 'The quick brown fox jumps hahaha over the quick dog' });
    store2.add({ Id: 4, Code: '4', Name: 'Brown fox hahaha brown dog' });
    store2.delete(3);

    expect(store1.search('hahaha')).toEqual(store2.search('hahaha'));
    expect(store2.search('hahaha')).toMatchSnapshot();
  });
});
