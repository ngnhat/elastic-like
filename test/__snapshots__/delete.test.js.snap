// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single field: multiple doc - single term 1`] = `
Array [
  Object {
    "score": 1.052597247134536,
    "source": Object {
      "Code": "4",
      "Id": 4,
      "Name": "Brown fox hahaha brown dog",
    },
  },
]
`;
